tck/src/main/java/org/eclipse/microprofile/lra/tck/TckLRATypeTests.java

```diff
@@ -0,0 +1,347 @@
+        if(tckSuiteClient != null) {
+            tckSuiteClient.close();
+        }
+    }
+
+    @Before
+    public void before() {
+        LOGGER.info("Running test: " + testName.getMethodName());
+        setUpTestCase();
+
+        try {
+            tckSuiteTarget = tckSuiteClient.target(URI.create(new URL(config.tckSuiteBaseUrl()).toExternalForm()));
+            lraClient = new LRAClientOps(tckSuiteTarget);
+        } catch (MalformedURLException mfe) {
+            throw new IllegalStateException("Cannot create URL for the LRA TCK suite base url " + config.tckSuiteBaseUrl(), mfe);
+        }
+    }
+
+    private void setUpTestCase() {
+        tckSuiteClient = ClientBuilder.newClient();
+    }
+
+    // enum to indicate which checks to perform on the expected and actual active LRA after running a resource method
+    private enum MethodLRACheck {
+        NONE, NOT_PRESENT, EQUALS, NOT_EQUALS
+    }
+
+    @Test
+    public void requiredWithLRATest() {
+        resourceRequest(REQUIRED_PATH, true, 200, MethodLRACheck.EQUALS, false);
+    }
+
+    @Test
+    public void requiredWithoutLRATest() {
+        resourceRequest(REQUIRED_PATH, false, 200, MethodLRACheck.NOT_EQUALS, false);
+    }
+
+    @Test
+    public void requiresNewWithLRATest() {
+        resourceRequest(REQUIRES_NEW_PATH, true, 200, MethodLRACheck.NOT_EQUALS, false);
+    }
+
+    @Test
+    public void requiresNewWithoutLRATest() {
+        resourceRequest(REQUIRES_NEW_PATH, false, 200, MethodLRACheck.NOT_EQUALS, false);
+    }
+
+    @Test
+    public void mandatoryWithLRATest() {
+        resourceRequest(MANDATORY_PATH, true, 200, MethodLRACheck.EQUALS, false);
+    }
+
+    @Test
+    public void mandatoryWithoutLRATest() {
+        resourceRequest(MANDATORY_PATH, false, 412, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    @Test
+    public void supportsWithLRATest() {
+        resourceRequest(SUPPORTS_PATH, true, 200, MethodLRACheck.EQUALS, false);
+    }
+
+    @Test
+    public void supportsWithoutLRATest() {
+        resourceRequest(SUPPORTS_PATH, false, 200, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    @Test
+    public void notSupportedWithRATest() {
+        resourceRequest(NOT_SUPPORTED_PATH, true, 200, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    @Test
+    public void notSupportedWithoutLRATest() {
+        resourceRequest(NOT_SUPPORTED_PATH, false, 200, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    @Test
+    public void neverWithRATest() {
+        resourceRequest(NEVER_PATH, true, 412, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    @Test
+    public void neverWithoutLRATest() {
+        resourceRequest(NEVER_PATH, false, 200, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    // same set of tests except that the invoked method end = true set on the LRA annotation
+
+    @Test
+    public void requiredEndWithLRATest() {
+        resourceRequest(REQUIRED_WITH_END_PATH, true, 200, MethodLRACheck.EQUALS, true);
```

Comment:
_WITH_END is confusing here. Because it is an endpoint were end=false is defined.